### DISCUSSION TITLE
How to access the options set in the SWRConfig?

### DISCUSSION THREAD

**Discussion by juvsqz** [vercel/swr#880] [Q&A]
Good day. I just want to ask how to access the options set in the **SWRConfig**?  It seems that **SWRConfigContext** is not exposed publicly.

I am planning to create a react hook that depends on the global options set in the **SWRConfig**

Thanks in advance!

**Comment by scucchiero** [comment]
Hey there!
For global options, would something like this work for you?

```
const [swrConfig] = useState({
    //your config here
  })
 
  //you could create your context here and wrap your app with it
const SWRContext = React.createContext(.........

return (
  <SWRContext>
    <SWRConfig value={swrConfig}>
      <App/>
    </SWRConfig>
  </SWRContext>
)
```

For local options, you could do something similar

**Reply by juvsqz** [reply]
Fair enough. So we will be needing to create our own context in order to populate these fields to the tree.  I hope, soon, they will expose a hook for this options/configs. 

Similar to this. `const { fetcher } = useSWRConfig();`

Thanks for the answer. Appreciated it a lot.
